src/components/implementation/tests/unit_torrent/torrent_test.c

```diff
@@ -6,116 +6,223 @@
-	char *data1 = "1234567890", *data2 = "asdf;lkj", *data3 = "asdf;lkj1234567890";
-	unsigned int ret1, ret2;
+	int chunk_size;
+	
+	/* open file */
+	t = tsplit(cos_spd_id(), td_root, params1, strlen(params1), TOR_ALL, evt1);
+	if (t < 1) { printc("UNIT TEST FAILED: split failed %c\n", t); return; }
+
+	for (chunk_size = 64; chunk_size <= 16384; chunk_size *= 2) {
+		unsigned int iterations = 5;
+		unsigned int max = iterations;
+		long long total = 0;
+
+		while (0 < iterations--) {
+			total += doit(t, chunk_size);
+		}
+		long long average = (long long) ((long double) total / (long double) max);
+		printc("avr clock ticks for chunk size %d: %lld\n", chunk_size, average);
+	}
 
-	int a = 0, b = 0, c = 0;
-	c = treadp(cos_spd_id(), 0, &a, &b);
+	/* close file */
+	trelease(cos_spd_id(), t);
 
-	printc("UNIT TEST Unit tests for torrents...\n");
+	call();
+}
 
-	printc("%d %d %d\n", a, b, c);
+void perf_tests(void)
+{
+	test_write();
+}
 
-	evt1 = evt_split(cos_spd_id(), 0, 0);
-	evt2 = evt_split(cos_spd_id(), 0, 0);
-	assert(evt1 > 0 && evt2 > 0);
+void twritep_readp_tests(void)
+{
+	td_t t1, t2;
+	long evt1 = 0;
+	char *params1 = "bar";
+	int ret1, ret2;
 
+	int a = 0, b = 0, c = 0;
+	
 	t1 = tsplit(cos_spd_id(), td_root, params1, strlen(params1), TOR_ALL, evt1);
-	if (t1 < 1) {
-		printc("UNIT TEST FAILED: split failed %d\n", t1);
-		return;
-	}
+	if (t1 < 1) { printc("UNIT TEST FAILED: split2 failed %c\n", t1); return; }
+	cbuf_t cb;
+	char *str1 = "AAA games are just great!";
+	char *d = cbuf_alloc(strlen(str1), &cb);
+	if (!d) { printc("UNIT TEST FAILED: split2 failed %c\n", *d); return; }
+	memcpy(d, str1, strlen(str1));
+	c = twritep(cos_spd_id(), t1, cb, 0, strlen(str1));
+	
+	cbuf_t cb2;
+	char *str2 = " Actually wait, no. What am I saying?";
+	char *e = cbuf_alloc(strlen(str2), &cb2);
+	if (!e) { printc("UNIT TEST FAILED: split2 failed %c\n", *e); return; }
+	memcpy(e, str2, strlen(str2));
+	c = twritep(cos_spd_id(), t1, cb2, 0, strlen(str2));
+
+	cbuf_t cb3;
+	char *str3 = "terrible!";
+	char *f = cbuf_alloc(strlen(str3), &cb3);
+	if (!f) { printc("UNIT TEST FAILED: split2 failed %c\n", *f); return; }
+	memcpy(f, str3, strlen(str3));
+	char val[8];
+	snprintf(val, 8, "%d", 14);
+	int rt = twmeta(cos_spd_id(), t1, "offset", strlen("offset"),
+		 val, strlen(val));
+	c = twritep(cos_spd_id(), t1, cb3, 0, strlen(str3));
+	
+	cbuf_t cb4;
+	char *str4 = " Diablo 1 and Torchlight are the best! Also Borderlands.";
+	char *g = cbuf_alloc(strlen(str4), &cb4);
+	if (!g) { printc("UNIT TEST FAILED: split2 failed %c\n", *g); return; }
+	memcpy(g, str4, strlen(str4));
+	snprintf(val, 8, "%d", 23);
+	rt = twmeta(cos_spd_id(), t1, "offset", strlen("offset"),
+		 val, strlen(val));
+	c = twritep(cos_spd_id(), t1, cb4, 0, strlen(str4));
+	
+	// Close the file so we can read from it
 	trelease(cos_spd_id(), t1);
-	printc("UNIT TEST PASSED: split->release\n");
 
-	t1 = tsplit(cos_spd_id(), td_root, params2, strlen(params2), TOR_ALL, evt1);
+
+
+
```

Comment:
if you want to separate out the logical functionalities, use separate functions.  Large amounts of whitespace is not a good answer for that.